File: Data Structures & Algorithms/k-closest-points-to-origin/submission-2.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<pair<double, pair<int, int>>> q;
        vector<vector<int>> res;
        for(auto p : points){
            double dis = p[0]*p[0] + p[1]*p[1];
            q.push({dis, {p[0], p[1]}});
            if(q.size() > k){
                q.pop();
            }

        }
        for(int i = 0; i < k; i++){
            res.push_back({q.top().second.first, q.top().second.second});
            q.pop();
        }
        return res;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-2.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0]*len(temperatures)
        stack = []
        for i in range(len(temperatures)):
            while stack and temperatures[i] > stack[-1][0]:
                j = stack[-1][1]
                res[stack.pop()[1]] = i - j
            stack.append((temperatures[i], i))
        return res
